### ISSUE TITLE
Module Setup Flow | AdSense: Existing user, no code on site

### ISSUE THREAD

**Issue by bjackson27** [google/site-kit-wp#269]
See https://docs.google.com/drawings/d/1OVlWdKgMNOjcE-btT5fJoO2FLq9E5C5rxnEj9rHeiM0/edit

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* User clicks “connect service” for AdSense 
* Requested to grant AdSense API scopes → yes
* Check if they have an account → account is detected and is already approved, but no code on the site.  
* Show screen “Looks like you’re already using AdSense” → user has the choice to “let Site Kit place code on your site” or not via a toggle. Toggle is *on* by default.

Scenario 1: User leaves toggle “let Site Kit place code” on and clicks "continue"
* Plugin places AdSense code on the site
* Check Impressions via reporting API:
    * if zero → Show screen “We’re getting your site ready for ads” + link to AdSense FE requesting user to add their site to Sites tab and enable ads 
    * if non-zero → show AdSense dashboard
* Settings page:
     * Account status: Approved
     * AdSense code: AdSense code is placed on your site

Scenario 2: User toggles off “let Site Kit place code” 
* Plugin DOES NOT place code on site 
* Check impressions via reporting API:
    * if zero → send user to Settings page 
    * if non-zero → then show AdSense dashboard. Redirect user to the AdSense dashboard within the plugin
* Settings page:
     * Account status: Approved
     * AdSense code: AdSense code is not placed on your site

## Implementation Brief

* n/a

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
@bjackson27 @aaemnnosttv I wonder whether we should combine #271 and #272 in this issue here and close them instead? Alternatively, we implement one of the two first and add the other one once the first one is merged - they shouldn't be worked on at the same time I feel, since that would likely produce conflicts.

**Comment by aaemnnosttv** [comment]
It seems like there may be some overlap between the issues. #272 provides much more information in terms of what should be tested I think, which is great. I feel like #271 needs more information.

Perhaps we should split the specific flows into their own issues? If we broke them down into separate smaller tests, it might make them easier to work on in parallel.

**Comment by felixarntz** [comment]
@aaemnnosttv Makes sense. Let's leave #271 as one, I believe that's fair because it only tests a similar flow with different types of accounts. #272 can be split into multiple ones, I'll review that with Mariya.

**Comment by aaemnnosttv** [comment]
@felixarntz @marrrmarrr 

Regarding the conditions in the criteria, can we remove the "if" conditions and make these into additional scenarios if necessary?

**Comment by marrrmarrr** [comment]
> @felixarntz @marrrmarrr
> 
> Regarding the conditions in the criteria, can we remove the "if" conditions and make these into additional scenarios if necessary?

Sure, that sounds good!

**Comment by marrrmarrr** [comment]
These conditions match the current user flows; let's revisit if we change the flows.

**Comment by felixarntz** [comment]
Closing in favor of #940
